cachito/workers/tasks/gomod.py

```diff
@@ -39,6 +40,27 @@ def _find_missing_gomod_files(bundle_dir, subpaths):
     return invalid_gomod_files
 
 
+def _is_workspace(repo_root: Path, subpath: str):
+    current_path = repo_root / subpath
+
+    while current_path != repo_root:
+        if (current_path / "go.work").exists():
+            return True
+        current_path = current_path.parent
+
+    if (repo_root / "go.work").exists():
+        return True
+
+    return False
+
+
+def _fail_if_bundle_dir_has_workspaces(bundle_dir: RequestBundleDir, subpaths: list[str]):
+    for subpath in subpaths:
+        if _is_workspace(bundle_dir.source_root_dir, subpath):
+            log.warning("go.work file found at %s", bundle_dir / subpath)
```

Comment:
go.work is not necessarily at `bundle_dir / subpath` - do we want to log the actual path or mention that it might be in  a parent dir?

Reply:
My bad, I realized my mistake afterwards and forgot to update the logs.

I think it's enough to log `bundle_dir.source_root_dir / subpath`, right? Subpath might only be a `.`, in which case, pathlib.Path will resolve the operation.

Reply:
Oh, I didn't notice the missing `source_root_dir`

My comment was about the message being misleading
```
go.work file found at .../repo/workspace/mod_a
```
But there's no go.work file there, it is at `.../repo/workspace`

Reply:
maybe just add `go.work file found at ... (or a parent directory)`

Reply:
You're right, I completely missed that.